File: DateTimeCheckerApp/ValidDate.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows;

namespace DateTimeCheckerApp
{
    public class ValidDate
    {
        public bool CheckDate(string dayStr, string monthStr, string yearStr)
        {
            string msg = "";

            
            int day, month, year;

            if (string.IsNullOrEmpty(dayStr) || string.IsNullOrEmpty(monthStr) || string.IsNullOrEmpty(yearStr))
            {
                msg = "Please enter all the date fields.";
                MessageBox.Show(msg, "Warning", MessageBoxButton.OK, MessageBoxImage.Warning);
                return false;
            }

            try
            {
                day = int.Parse(dayStr);
                month = int.Parse(monthStr);
                year = int.Parse(yearStr);
            }
            catch (Exception ex)
            {
                MessageBox.Show("Invalid input format.", "Error", MessageBoxButton.OK, MessageBoxImage.Error);
                return false;
            }
            if (day < 1 || day > 31)
            {
                msg = "Input data for Day is out of range";
                MessageBox.Show(msg, "Warning", MessageBoxButton.OK, MessageBoxImage.Error);
                return false;
            }
            if (month < 1 || month > 12)
            {
                msg = "Input data for Month is out of range";
                MessageBox.Show(msg, "Warning", MessageBoxButton.OK, MessageBoxImage.Error);
                return false;
            }
            if (year < 1000 || year > 3000)
            {
                msg = "Input data for Year is out of range";
                MessageBox.Show(msg, "Warning", MessageBoxButton.OK, MessageBoxImage.Error);
                return false;
            }

            //DateTime.Parse($"{day}/{month}/{year}");
            //MessageBox.Show($"{day}/{month}/{year} is correct date time!", "Message", MessageBoxButton.OK, MessageBoxImage.Information);
            //return true;
            try
            {
                monthStr = month.ToString();
                yearStr = year.ToString();

                DayInMonth dayInMonth = new DayInMonth();
                int daysInMonth = dayInMonth.CheckDayInMonth(monthStr, yearStr);

                if (daysInMonth == -1)
                {
                    return false; // Invalid month
                }
                else if (day < 1 || day > daysInMonth)
                {
                    MessageBox.Show($"{day}/{month}/{year} is not correct date time!", "Message", MessageBoxButton.OK, MessageBoxImage.Information);
                    return false; // Invalid day
                }
                else
                {
                    MessageBox.Show($"{day}/{month}/{year} is correct date time!", "Message", MessageBoxButton.OK, MessageBoxImage.Information);
                    return true; // Valid date
                }
            }
            catch (Exception ex)
            {
                
                MessageBox.Show($"{day}/{month}/{year} is not correct date time!", "Message", MessageBoxButton.OK, MessageBoxImage.Information);
                return false;
            }
        }
    }
}

File: DateTimeCheckerAppPython/ValidDate.py
import tkinter as tk
from tkinter import messagebox
from DayInMonth import DayInMonth


class ValidDate:
    def __init__(self): self.day_in_month = DayInMonth()
    def check_date(self, day_str, month_str, year_str):
        try:
            day = int(day_str)
            month = int(month_str)
            year = int(year_str)
        except ValueError:
            messagebox.showerror("Invalid Input", "Invalid input format.")
            return False
        
        if day < 1 or day > 31:
            messagebox.showerror("Invalid Input", "Input data for Day is out of range")
            return False
        if month < 1 or month > 12:
            messagebox.showerror("Invalid Input", "Input data for Month is out of range")
            return False
        if year < 1000 or year > 3000:
            messagebox.showerror("Invalid Input", "Input data for Year is out of range")
            return False
        
        try:
            day_in_month = DayInMonth().check_day_in_month(month_str, year_str)
            if day_in_month == -1:
                return False  # Invalid month
            elif day < 1 or day > day_in_month:
                messagebox.showerror("Invalid Date", f"{day}/{month}/{year} is not correct date time!")
                return False  # Invalid day
            else:
                messagebox.showinfo("Valid Date", f"{day}/{month}/{year} is correct date time!")
                return True  # Valid date
        except Exception as e:
            messagebox.showerror("Invalid Date", f"{day}/{month}/{year} is not correct date time!")
            return False




